### DISCUSSION TITLE
Add third, undeterminated state to the result labels

### DISCUSSION THREAD

**Discussion by mellowagain** [autometrics-dev/autometrics-shared#95] [Ideas]
> Migrated from https://github.com/autometrics-dev/autometrics-shared/pull/85#discussion_r1348401386

In the spec draft v1.0.0 only two result label values are accepted: `ok` and `error`

My suggestions would be to have a third, undeterminated state (e.g `n/a`). This would allow consumers of the library (e.g Dora) to:

- Guide users to add correct function success/failure, if the library fails to automatically do so
- Warn users if they have SLOs that depend on these functions

This would necessitate a breaking change in the spec as the current spec draft requires functions, for which a result cannot be determinated, to be labeled as `ok`.

**Comment by flenter** [comment]
This might not be the best place for this, but I'm more struggling with coming up with cases where autometrics would be unable to determine whether something was executed successfully. What sort of situations would that be?
